### ISSUE TITLE
Can use pretrain model for calc perplexity ?

### ISSUE THREAD

**Issue by loserdog-err** [dbiir/UER-py#33]
I want to score a sentence by calc the perplexity.
Can display some sample code or give some  ideas?
Thanks~

**Comment by zhezhaoa** [comment]
I think you can achieve this by changing some codes in scripts/generate.py 
`
for i in range(args.seq_length):
    outputs = model(src_tensor, seg_tensor)
    next_token_logits = outputs[0][-1]
    prob = F.softmax(next_token_logits, dim=-1)
`
prob is the distribution of the next token.
Then you can obtain the PPL easily.

**Comment by loserdog-err** [comment]
> I think you can achieve this by changing some codes in scripts/generate.py
> `for i in range(args.seq_length): `
> ` outputs = model(src_tensor, seg_tensor)`
> ` next_token_logits = outputs[0][-1]`
> ` prob = F.softmax(next_token_logits, dim=-1)`
> 
> prob is the distribution of the next token.
> Then you can obtain the PPL easily.

感谢回复。
请问当我输入长度为2的文本之后，第一个循环的 outputs shape 为 (1,2,21128) ,经过 softmax 后，代表的就是我输入这两个字的概率分布吗？

**Comment by zhezhaoa** [comment]
代表的是根据当前位置的字预测下一个字的概率分布。然后需要知道真实的下一个字是什么，得到这个字的概率（从前面提到的分布中），从而计算PPL

**Comment by loserdog-err** [comment]
> 代表的是根据当前位置的字预测下一个字的概率分布。然后需要知道真实的下一个字是什么，得到这个字的概率（从前面提到的分布中），从而计算PPL

ok，感谢~
